### ISSUE TITLE
Make p4.client a required field when using p4-fusion

### ISSUE THREAD

**Issue by marcleblanc2** [sourcegraph/sourcegraph#50145]
- **Sourcegraph version:** 5.0.0
- **Platform information:** p4-fusion

#### Steps to reproduce:
1. Leave p4.client blank
2. Save the code host config

#### Expected behavior:
1. Warning that p4.client is blank 

#### Actual behavior:
1. No warning
2. Perforce clone fails inelegantly

Client needs to be a required field when using p4-fusion.
https://sourcegraph.com/github.com/sourcegraph/sourcegraph@c9b78bcf3e0de34d26ae63d99bdb146434e1ec25/-/blob/cmd/gitserver/server/vcs_syncer_perforce.go?L136

Thread with [customer 6716](https://github.com/sourcegraph/accounts/issues/6716)
https://sourcegraph.slack.com/archives/C03R25C9WE6/p1675945533974859?thread_ts=1659016200.031759&cid=C03R25C9WE6

/cc @ryphil @mrnugget

**Comment by peterguy** [comment]
I've been cloning Perforce depots on our dev Sourcegraph instances for a month or two now. I have never set `p4.client` and I have never had any problems cloning depots. What am I missing?

**Comment by marcleblanc2** [comment]
Your Perforce server is creating a default client spec for you. Let’s book some time tomorrow and go through this. It is key. Regards,MarcOn Mar 30, 2023, at 20:23, Peter Guy ***@***.***> wrote:﻿
I've been cloning Perforce depots on our dev Sourcegraph instances for a month or two now. I have never set p4.client and I have never had any problems cloning depots. What am I missing?

**Comment by mrnugget** [comment]
Please leave a summary in this ticket.

**Comment by peterguy** [comment]
We ran out of time, but we confirmed that `p4-fusion` uses a default client spec (which comes from the Perforce server?) if no client is provided.

The `p4-fusion` command line is built in [vcs_syncer_perforce.go](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@ab7125eca10674ec77a1348f2bdeadd41e9d0eb4/-/blob/cmd/gitserver/server/vcs_syncer_perforce.go?L134-148).

It includes `--client` and a value from the config (`s.FusionConfig.Client`), but if that config parameter is empty, Golang's [os/exec Cmd struct](https://pkg.go.dev/os/exec#Cmd) quotes it so that the command line options are not disrupted.

`p4-fusion` receives `--client ""` when the code host config in Sourcegraph does not contain `"p4.client"`.

The cloning process with a blank/default client completes just fine from a user's perspective. It does not benefit from the settings that may be in a client spec, such as explicitly blocking files (@marcleblanc2 explained that `p4-fusion`'s binary file exclusion depends on Perforce/Helix's binary file detection, which determines if a file is a binary by looking for non-printable characters in the first 1,000 bytes of the file, which could miss files that actually are binary), but it clones the depot successfully.

I'm not sure this should be an issue. If there is no need for the extra functionality of a client spec, simply omit `"p4.client"`. If there are benefits to be had from using a client spec, `"p4.client"` is available.
